libraries/aws_ec2_instance.rb

```diff
@@ -85,6 +87,14 @@ def to_s
     "EC2 Instance #{@display_name}"
   end
 
+  def has_roles?
+    name = instance.iam_instance_profile.arn.gsub(%r{^.*\/}, '')
```

Comment:
I thought we had some nil checks here, but they're gone.

For example, what if the instance has no instance profile?